cmake/llext-edk.cmake

```diff
@@ -0,0 +1,106 @@
+    "${LLEXT_CFLAGS} -imacros\${${install_dir_var}}/include/zephyr/${autoconf_h_rel}")
+
+file(MAKE_DIRECTORY ${LLEXT_EDK_INC})
+foreach(dir ${INCLUDE_DIRS})
+    if (NOT EXISTS ${dir})
+        continue()
+    endif()
+    cmake_path(IS_PREFIX PROJECT_BINARY_DIR ${dir} NORMALIZE to_prj_bindir)
+    cmake_path(IS_PREFIX ZEPHYR_BASE ${dir} NORMALIZE to_zephyr_base)
+    cmake_path(IS_PREFIX WEST_TOPDIR ${dir} NORMALIZE to_west_topdir)
+    cmake_path(IS_PREFIX APPLICATION_SOURCE_DIR ${dir} NORMALIZE to_app_srcdir)
+
+    # Overall idea is to place included files in the destination dir based on the source:
+    # files coming from build/zephyr/generated will end up at
+    # <install-dir>/include/zephyr/include/generated, files coming from zephyr base at
+    # <install-dir>/include/zephyr/include, files from west top dir (for instance, hal modules),
+    # at <install-dir>/include and application ones at <install-dir>/include/<application-dir>.
+    # Finally, everything else (such as external libs not at any of those places) will end up
+    # at <install-dir>/include/<full-path-to-external-include>, so we avoid any external lib
+    # stepping at any other lib toes.
+    if(to_prj_bindir)
+        cmake_path(RELATIVE_PATH dir BASE_DIRECTORY ${PROJECT_BINARY_DIR} OUTPUT_VARIABLE dir_tmp)
+        set(dest ${LLEXT_EDK_INC}/zephyr/${dir_tmp})
+    elseif(to_zephyr_base)
+        cmake_path(RELATIVE_PATH dir BASE_DIRECTORY ${ZEPHYR_BASE} OUTPUT_VARIABLE dir_tmp)
+        set(dest ${LLEXT_EDK_INC}/zephyr/${dir_tmp})
+    elseif(to_west_topdir)
+        cmake_path(RELATIVE_PATH dir BASE_DIRECTORY ${WEST_TOPDIR} OUTPUT_VARIABLE dir_tmp)
+        set(dest ${LLEXT_EDK_INC}/${dir_tmp})
+    elseif(to_app_srcdir)
+        cmake_path(GET APPLICATION_SOURCE_DIR FILENAME app_dir)
+        cmake_path(RELATIVE_PATH dir BASE_DIRECTORY ${APPLICATION_SOURCE_DIR} OUTPUT_VARIABLE dir_tmp)
+        set(dest ${LLEXT_EDK_INC}/${app_dir}/${dir_tmp})
+    else()
+        set(dest ${LLEXT_EDK_INC}/${dir})
+    endif()
+
+    # Use destination parent, as the last part of the source directory is copied as well
+    cmake_path(GET dest PARENT_PATH dest_p)
+
+    file(MAKE_DIRECTORY ${dest_p})
+    file(COPY ${dir} DESTINATION ${dest_p} FILES_MATCHING PATTERN "*.h")
+
+    cmake_path(RELATIVE_PATH dest BASE_DIRECTORY ${LLEXT_EDK} OUTPUT_VARIABLE dest_rel)
+    if(to_prj_bindir)
+        list(APPEND inc_gen_flags_make "-I\$(${install_dir_var})/${dest_rel}")
+        list(APPEND inc_gen_flags_cmake "-I\${${install_dir_var}}/${dest_rel}")
+    else(to_zephyr_base)
+        list(APPEND inc_flags_make "-I\$(${install_dir_var})/${dest_rel}")
+        list(APPEND inc_flags_cmake "-I\${${install_dir_var}}/${dest_rel}")
+    endif()
+endforeach()
+
+# Generate flags for Makefile
+list(APPEND all_flags_make ${base_flags_make} ${inc_gen_flags_make} ${inc_flags_make})
+list(JOIN all_flags_make " " all_flags_str)
+file(WRITE ${LLEXT_EDK}/Makefile.cflags "LLEXT_CFLAGS = ${all_flags_str}")
```

Comment:
Have you considered using the approach used by the [makefile_exports](https://github.com/zephyrproject-rtos/zephyr/blob/main/cmake/makefile_exports/CMakeLists.txt) target? I think that's clearer and also should allow greater flexibility in generating the flag lists.

Reply:
Yep, that was my first try - it become something completely different because I wanted to trim details from the environment where the EDK is built: as the CFLAGS have absolute paths, they would leek into the EDK. That's why there's the "preprocessing" bit before, removing those. Thoughts?